source/postprocess/visualization.cc

```diff
@@ -251,8 +251,14 @@ namespace aspect
 
       // now also generate a .pvd file that matches simulation
       // time and corresponding .pvtu record
-      times_and_pvtu_names.push_back(std::make_pair
-                                     (time_in_years_or_seconds, "solution/"+pvtu_master_filename));
+      // in case we output all nonlinear iterations, we only want one
+      // entry per time step, so replace the last line with the current iteration
+      if (this->get_nonlinear_iteration() == 0 || !this->get_parameters().run_postprocessors_on_nonlinear_iterations)
```

Comment:
I find this logic hard to understand (it is correct though). I would prefer if you could make it more explicit on the cost of a few more lines, namely:
```
if (this->get_parameters().run_postprocessors_on_nonlinear_iterations)
{
if (this->get_nonlinear_iteration() == 0)
... push_back
else
... replace back
}
else
push_back
```